android/expoview/src/main/java/host/exp/exponent/notifications/schedulers/IntervalScheduler.java

```diff
@@ -0,0 +1,240 @@
+package host.exp.exponent.notifications.schedulers;
+
+import android.content.Context;
+import android.content.Intent;
+import android.os.SystemClock;
+
+import com.raizlabs.android.dbflow.annotation.Column;
+import com.raizlabs.android.dbflow.annotation.PrimaryKey;
+import com.raizlabs.android.dbflow.annotation.Table;
+import com.raizlabs.android.dbflow.structure.BaseModel;
+
+import org.joda.time.DateTime;
+import org.json.JSONException;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+
+import host.exp.exponent.notifications.ExponentNotificationManager;
+import host.exp.exponent.notifications.exceptions.UnableToScheduleException;
+import host.exp.exponent.notifications.managers.SchedulersManagerProxy;
+import host.exp.exponent.notifications.interfaces.SchedulerInterface;
+import host.exp.exponent.notifications.managers.SchedulersDatabase;
+
+@Table(databaseName = SchedulersDatabase.NAME)
+public class IntervalScheduler extends BaseModel implements SchedulerInterface {
+
+  private List<String> mTriggeringActions = Arrays.asList(null,
+      Intent.ACTION_REBOOT,
+      Intent.ACTION_BOOT_COMPLETED);
+
+  private Context mApplicationContext;
+
+  public HashMap<String, Object> details;
+
+  // -- model fields --
+
+  @Column
+  @PrimaryKey
+  int id;
+
+  @Column
+  int notificationId;
+
+  @Column
+  String experienceId;
+
+  @Column
+  boolean repeat;
+
+  @Column
+  boolean scheduled = false;
+
+  @Column
+  String serializedDetails;
+
+  @Column
+  long scheduledTime;
+
+  @Column
+  long interval;
+
+  // -- scheduler methods --
+
+  @Override
+  public void setApplicationContext(Context context) {
+    mApplicationContext = context.getApplicationContext();
+  }
+
+  @Override
+  public void schedule(String action) throws UnableToScheduleException {
+    if (!mTriggeringActions.contains(action)) {
+      return;
+    }
+    long nextAppearanceTime = 0;
+
+    try {
+      nextAppearanceTime = getNextAppearanceTime();
+    } catch (IllegalArgumentException e) {
+      throw new UnableToScheduleException();
+    }
+
+    ensureDetails();
+    try {
+      getManager().schedule(experienceId, notificationId, details, nextAppearanceTime, null);
+    } catch (ClassNotFoundException e) {
```

Comment:
In what situation this exception can be thrown?

Reply:
This exception is in method signature.
`public void schedule(String experienceId, int id, HashMap details, long time, Long interval) throws ClassNotFoundException {`